### ISSUE TITLE
rec_train.py训练时准确率一直为0

### ISSUE THREAD

**Issue by ZZHHogan** [WenmuZhou/PytorchOCR#111]
你好，我这里使用icdar 2017数据集训练，发现准确率一直为0，使用的配置项为rec_train_config.py，没有做什么太大的修改，只是修改了batch_size，alphabet的路径，dataset的路径，训练时log的打印如下：
![image](https://user-images.githubusercontent.com/40531744/98208912-0fb5ab00-1f79-11eb-8e38-e7bc4364f5f0.png)
期间进入到RecMetric.py中查看模型的输出值predictions，发现里面预测的结果都为0,0,0...0的tensor

训练数据的标注如下（图片路径与文字内容中间以\t进行分割）：

E:\DataSets\icdar2017rctw\icdar2017rctw\recognition\train\image_0_0.jpg	金氏眼镜
E:\DataSets\icdar2017rctw\icdar2017rctw\recognition\train\image_0_1.jpg	创于1989
E:\DataSets\icdar2017rctw\icdar2017rctw\recognition\train\image_0_2.jpg	城建店

alphabet中的信息如下，一共5529个字符，加上blank的话那么最后的类别数n_class设置为5530：
![image](https://user-images.githubusercontent.com/40531744/98208700-b0f03180-1f78-11eb-8573-6c6bef8ac0c5.png)

另外config中的设置项：
2020-11-05 15:10:33,337 - torchocr - INFO - {'exp_name': 'CRNN', 'train_options': {'resume_from': '', 'third_party_name': '', 'checkpoint_save_dir': './output/CRNN/checkpoint', 'device': 'cuda:0', 'epochs': 200, 'fine_tune_stage': ['backbone', 'neck', 'head'], 'print_interval': 20, 'val_interval': 3000, 'ckpt_save_type': 'HighestAcc', 'ckpt_save_epoch': 4}, 'SEED': 927, 'optimizer': {'type': 'Adam', 'lr': 0.001, 'weight_decay': 0.0001}, 'lr_scheduler': {'type': 'StepLR', 'step_size': 60, 'gamma': 0.1}, 'model': {'type': 'RecModel', 'backbone': {'type': 'ResNet', 'layers': 18}, 'neck': {'type': 'PPaddleRNN'}, 'head': {'type': 'CTC', 'n_class': 5530}, 'in_channels': 3}, 'loss': {'type': 'CTCLoss', 'blank_idx': 0}, 'dataset': {'alphabet': 'E:/pro/ncnn_ocr/models/keys.txt', 'train': {'dataset': {'type': 'RecTextLineDataset', 'file': 'E:/pro/chineseocr-master/train/ocr/txt/icdar2017Backup.txt', 'input_h': 32, 'mean': 0.5, 'std': 0.5, 'augmentation': False}, 'loader': {'type': 'DataLoader', 'batch_size': 4, 'shuffle': True, 'num_workers': 1, 'collate_fn': {'type': 'RecCollateFn', 'img_w': 120}}}, 'eval': {'dataset': {'type': 'RecTextLineDataset', 'file': 'E:/pro/chineseocr-master/train/ocr/txt/2017valBackup.txt', 'input_h': 32, 'mean': 0.5, 'std': 0.5, 'augmentation': False}, 'loader': {'type': 'RecDataLoader', 'batch_size': 4, 'shuffle': False, 'num_workers': 1, 'collate_fn': {'type': 'RecCollateFn', 'img_w': 120}}}}}

希望得到回复，谢谢！

**Comment by terrancraft** [comment]
你打印一下网络结构看看对不对

**Comment by novioleo** [comment]
@hhhhhhogan 抱歉，才看到这个issue。能烦请你上传一下你的对应的图片，以及你的onehot的结果么？

**Comment by cendelian** [comment]
> 你好，我这里使用icdar 2017数据集训练，发现准确率一直为0，使用的配置项为rec_train_config.py，没有做什么太大的修改，只是修改了batch_size，alphabet的路径，dataset的路径，训练时log的打印如下：
> ![image](https://user-images.githubusercontent.com/40531744/98208912-0fb5ab00-1f79-11eb-8e38-e7bc4364f5f0.png)
> 期间进入到RecMetric.py中查看模型的输出值predictions，发现里面预测的结果都为0,0,0...0的tensor
> 
> 训练数据的标注如下（图片路径与文字内容中间以\t进行分割）：
> 
> E:\DataSets\icdar2017rctw\icdar2017rctw\recognition\train\image_0_0.jpg 金氏眼镜
> E:\DataSets\icdar2017rctw\icdar2017rctw\recognition\train\image_0_1.jpg 创于1989
> E:\DataSets\icdar2017rctw\icdar2017rctw\recognition\train\image_0_2.jpg 城建店
> 
> alphabet中的信息如下，一共5529个字符，加上blank的话那么最后的类别数n_class设置为5530：
> ![image](https://user-images.githubusercontent.com/40531744/98208700-b0f03180-1f78-11eb-8573-6c6bef8ac0c5.png)
> 
> 另外config中的设置项：
> 2020-11-05 15:10:33,337 - torchocr - INFO - {'exp_name': 'CRNN', 'train_options': {'resume_from': '', 'third_party_name': '', 'checkpoint_save_dir': './output/CRNN/checkpoint', 'device': 'cuda:0', 'epochs': 200, 'fine_tune_stage': ['backbone', 'neck', 'head'], 'print_interval': 20, 'val_interval': 3000, 'ckpt_save_type': 'HighestAcc', 'ckpt_save_epoch': 4}, 'SEED': 927, 'optimizer': {'type': 'Adam', 'lr': 0.001, 'weight_decay': 0.0001}, 'lr_scheduler': {'type': 'StepLR', 'step_size': 60, 'gamma': 0.1}, 'model': {'type': 'RecModel', 'backbone': {'type': 'ResNet', 'layers': 18}, 'neck': {'type': 'PPaddleRNN'}, 'head': {'type': 'CTC', 'n_class': 5530}, 'in_channels': 3}, 'loss': {'type': 'CTCLoss', 'blank_idx': 0}, 'dataset': {'alphabet': 'E:/pro/ncnn_ocr/models/keys.txt', 'train': {'dataset': {'type': 'RecTextLineDataset', 'file': 'E:/pro/chineseocr-master/train/ocr/txt/icdar2017Backup.txt', 'input_h': 32, 'mean': 0.5, 'std': 0.5, 'augmentation': False}, 'loader': {'type': 'DataLoader', 'batch_size': 4, 'shuffle': True, 'num_workers': 1, 'collate_fn': {'type': 'RecCollateFn', 'img_w': 120}}}, 'eval': {'dataset': {'type': 'RecTextLineDataset', 'file': 'E:/pro/chineseocr-master/train/ocr/txt/2017valBackup.txt', 'input_h': 32, 'mean': 0.5, 'std': 0.5, 'augmentation': False}, 'loader': {'type': 'RecDataLoader', 'batch_size': 4, 'shuffle': False, 'num_workers': 1, 'collate_fn': {'type': 'RecCollateFn', 'img_w': 120}}}}}
> 
> 希望得到回复，谢谢！

你好，，请问你解决这个问题了吗？我也遇到了这个问题

**Comment by luotianhang** [comment]
您好，我也遇到了这个问题，请问您是怎么解决的

**Comment by wzl639** [comment]
我也是这个问题

**Comment by novioleo** [comment]
重新拉取代码，并升级下pytorch的版本，到1.7试试。
@wzl639

**Comment by wzl639** [comment]
> 重新拉取代码，并升级下pytorch的版本，到1.7试试。
> @wzl639
感谢解答 但是我的torch是1.7

**Comment by wzl639** [comment]
> > 重新拉取代码，并升级下pytorch的版本，到1.7试试。
> > @wzl639
> > 感谢解答 但是我的torch是1.7

博主我还有一个问题 单独发出来 能帮我看看吗

**Comment by luotianhang** [comment]
我做过实验 你要训练到很后面才会开始有效果

**Comment by Tomhardy13** [comment]
> 我做过实验 你要训练到很后面才会开始有效果

请问一般要多久才开始有效果？@luotianhang

**Comment by fourierer** [comment]
> > 我做过实验 你要训练到很后面才会开始有效果
> 
> 请问一般要多久才开始有效果？@luotianhang

我尝试了icdar2015，基本上要train到第20-30个epoch才有效果
